draft-ietf-quic-tls.md

```diff
@@ -1024,144 +795,90 @@ encrypted_pn = ChaCha20(pn_key, counter, nonce, packet_number)
 
-Even though newer keys could be available when retransmitting, retransmissions
-of these handshake messages MUST be sent in packets protected with handshake
-keys.  An endpoint MUST generate ACK frames for these messages and send them in
-packets protected with handshake keys.
+## Receiving Out-of-Order Protected Frames {#pre-hs-protected}
 
-A HelloRetryRequest handshake message might be used to reject an initial
-ClientHello.  A HelloRetryRequest handshake message is sent in a Retry packet;
-any second ClientHello that is sent in response uses a Initial packet type.
-These packets are only protected with a predictable key (see
-{{handshake-secrets}}).  This is natural, because no shared secret will be
-available when these messages need to be sent.  Upon receipt of a
-HelloRetryRequest, a client SHOULD cease any transmission of 0-RTT data; 0-RTT
-data will only be discarded by any server that sends a HelloRetryRequest.
+Due to reordering and loss, protected packets might be received by an endpoint
+before the final TLS handshake messages are received.  A client will be unable
+to decrypt 1-RTT packets from the server, whereas a server will be able to
+decrypt 1-RTT packets from the client. However,
+a server MUST NOT process data from incoming 1-RTT protected packets
+before verifying either the client Finished message or - in the case that the
+server has chosen to use a pre-shared key - the pre-shared key binder (see
+Section 4.2.8 of {{!TLS13}}).  Verifying these values provides the server with
+an assurance that the ClientHello has not been modified.
+Packets protected with 1-RTT keys MAY be stored and later decrypted and used
+once the handshake is complete.
 
-The packet type ensures that protected packets are clearly distinguished from
-unprotected packets.  Loss or reordering might cause unprotected packets to
-arrive once 1-RTT keys are in use, unprotected packets are easily distinguished
-from 1-RTT packets using the packet type.
+A server could receive packets protected with 0-RTT keys prior to receiving a
+TLS ClientHello.  The server MAY retain these packets for later decryption in
+anticipation of receiving a ClientHello.
 
-Once 1-RTT keys are available to an endpoint, it no longer needs the TLS
-handshake messages that are carried in unprotected packets.  However, a server
-might need to retransmit its TLS handshake messages in response to receiving an
-unprotected packet that contains ACK frames.  A server MUST process ACK frames
-in unprotected packets until the TLS handshake is reported as complete, or it
-receives an ACK frame in a protected packet that acknowledges all of its
-handshake messages.
+Receiving and verifying the TLS Finished message is critical in
+ensuring the integrity of the TLS handshake.  A server MUST NOT use
+1-RTT protected packets from the client prior to verifying the client
+Finished message if its response depends on client authentication.
```

Comment:
This paragraph seems redundant. The PR already has the following text starting from line 843,
```
However,
a server MUST NOT process data from incoming 1-RTT protected packets
before verifying either the client Finished message or - in the case that the
server has chosen to use a pre-shared key - the pre-shared key binder (see
Section 4.2.8 of {{!TLS13}}).
```